File: DiplomskiAndroid/app/src/main/java/com/example/diplomskiandroid/models/Vehicle.java
package com.example.diplomskiandroid.models;

public class Vehicle {
    private int id;
    private int userId;
    private String brand;
    private String model;
    private int year;
    private String licensePlate;
    private String vin;
    private String fuelType;
    private int mileage;
    private String registrationDate;
    private String note;
    private String createdAt;

    public int getId() { return id; }
    public int getUserId() { return userId; }
    public String getBrand() { return brand; }
    public String getModel() { return model; }
    public int getYear() { return year; }
    public String getLicensePlate() { return licensePlate; }
    public String getVin() { return vin; }
    public String getFuelType() { return fuelType; }
    public int getMileage() { return mileage; }
    public String getRegistrationDate() { return registrationDate; }
    public String getNote() { return note; }
    public String getCreatedAt() { return createdAt; }
}

File: DiplomskiAPI/Models/Vehicle.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DiplomskiAPI.Models
{
    [Table("vehicles")]
    public class Vehicle
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Column("user_id")]
        public int UserId { get; set; }

        [Column("brand")]
        public string Brand { get; set; } = string.Empty;

        [Column("model")]
        public string Model { get; set; } = string.Empty;

        [Column("year")]
        public int Year { get; set; }

        [Column("license_plate")]
        public string? LicensePlate { get; set; }

        [Column("vin")] //šasija
        public string? Vin { get; set; }

        [Column("fuel_type")]
        public string? FuelType { get; set; }

        [Column("mileage")]
        public int Mileage { get; set; }

        [Column("registration_date")]
        public DateTime? RegistrationDate { get; set; }

        [Column("note")]
        public string? Note { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; }

        [ForeignKey(nameof(UserId))]
        public User User { get; set; } = null!;
        [Column("is_active")]
        public bool IsActive { get; set; } = true;

        public ICollection<ServiceRequest> ServiceRequests { get; set; } = new List<ServiceRequest>();
    }
}